src/main/java/frc/robot/CommandFactory.java

```diff
@@ -37,6 +40,65 @@ public CommandFactory(
         this.superStructure = superStructure;
     }
 
+    private boolean atTargetPosition(double currentPosition, double targetPosition, double threshold) {
+        return Math.abs(currentPosition - targetPosition) <= threshold;
+    }
+
+    /**
+     * Tells the super Structure where to go based on the Robot Mode
+     *
+     * @param robotMode A supplier for the current RobotMode so it knows where to go
+     * @return The command to follow the current position based on the Robot Mode
+     */
+    public Command goToPosition(Supplier<RobotMode> robotMode) {
+        DoubleSupplier targetRobotModeElevatorHeight = () -> switch (robotMode.get()) {
+            case CORAL_LEVEL_1 -> Constants.ArmPositions.CORAL_LEVEL_1_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_2 -> Constants.ArmPositions.CORAL_LEVEL_2_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_3 -> Constants.ArmPositions.CORAL_LEVEL_3_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_4 -> Constants.ArmPositions.CORAL_LEVEL_4_ELEVATOR_HEIGHT;
+            case ALGAE_PROCESSOR -> Constants.ArmPositions.ALGAE_PROCESSOR_ELEVATOR_HEIGHT;
+            case ALGAE_NET -> Constants.ArmPositions.ALGAE_NET_ELEVATOR_HEIGHT;
+            case ALGAE_REMOVE_LOWER -> Constants.ArmPositions.REMOVE_ALGAE_LOWER_ELEVATOR_HEIGHT;
+            case ALGAE_REMOVE_UPPER -> Constants.ArmPositions.REMOVE_ALGAE_UPPER_ELEVATOR_HEIGHT;
+        };
+        DoubleSupplier targetElevatorHeight = () -> {
+            if (atTargetPosition(elevator.getPosition(), targetRobotModeElevatorHeight.getAsDouble(), Elevator.POSITION_THRESHOLD * 2) ||
+                    atTargetPosition(arm.getShoulderPosition(), Constants.ArmPositions.SHOULDER_SAFE_ANGLE, Constants.ArmPositions.SHOULDER_SAFE_THRESHOLD))
+                return targetRobotModeElevatorHeight.getAsDouble();
+            else return elevator.getPosition();
+        };
+        DoubleSupplier targetShoulderAngle = () -> {
+            if (!atTargetPosition(elevator.getPosition(), targetRobotModeElevatorHeight.getAsDouble(), Elevator.POSITION_THRESHOLD))
+                return Constants.ArmPositions.SHOULDER_SAFE_ANGLE;
+            switch (robotMode.get()) {
```

Comment:
Technically we only need to go to the safe angle if the target shoulder angle is less than the safe angle.